src/classes/LookupFieldMapper.cls

```diff
@@ -0,0 +1,106 @@
+    protected LookupFieldMapper() {}
+
+    /**********************************************************************************
+    * @description Static method to get the current instance for Singleton pattern
+    * @return The instance of LookupFieldMapper.
+    **********************************************************************************/
+    public static LookupFieldMapper getInstance() {
+        if (instance == null) {
+            instance = new LookupFieldMapper();
+        }
+
+        return instance;
+    }
+
+    /**********************************************************************************************************
+    * @description Retrieve a map of LookupField models representing fields on a specified owner SObject type
+    * that is a lookup to a specified target SObject type by field name.
+    * @param  sObjectType  The SObject type where the field originates
+    * @param  lookupToSObjectType The SObject type where the field lookup targets
+    * @return Map<String, LookupFieldModel>
+    **********************************************************************************************************/ 
+    public virtual Map<String, LookupFieldModel> getModelByNameMapForSObjectType(SObjectType sObjectType, SObjectType lookupToSObjectType) {
+        Map<String, LookupFieldModel> lookupFieldModelByName = new Map<String, LookupFieldModel>();
+        Map<String, Schema.SObjectField> sObjectFieldsByName = UTIL_Describe.getAllFieldsForSObject(sObjectType.getDescribe().name);
+
+        for (Schema.SObjectField field : sObjectFieldsByName.values()){
+            DescribeFieldResult cachedFieldDescribe = field.getDescribe();
+            Boolean isLookupToTargetSObjectType = UTIL_Describe.isLookupFieldTo(cachedFieldDescribe, lookupToSObjectType);
+
+            if (isLookupToTargetSObjectType == true) {
+                LookupFieldModel lookupFieldModel = new LookupFieldModel(cachedFieldDescribe.getName(), cachedFieldDescribe.getLabel());
+                lookupFieldModelByName.put(lookupFieldModel.name, lookupFieldModel);
+            }
+        }
+
+        return lookupFieldModelByName;
+    }
+
+    /**********************************************************************************************************
+    * @description Retrieve a map of LookupField models representing fields on a specified owner SObject type
+    * that is a lookup to a specified target SObject type by field label.
+    * @param  sObjectType  The SObject type where the field originates
+    * @param  lookupToSObjectType The SObject type where the field lookup targets
+    * @return Map<String, LookupFieldModel>
+    **********************************************************************************************************/ 
+    public virtual Map<String, LookupFieldModel> getModelByLabelMapForSObjectType(SObjectType sObjectType, SObjectType lookupToSObjectType) {
```

Comment:
I would prefer we abandon this method and swap it out for method in the service that effectively does the following:

- Arguments: Map<String,LookupFieldModel>
- Returns: Map<String,LookupFieldModel>
- Intialize a return map Map<String,LookupFieldModel>
- For each LookupFieldModel in the map's values
-- add the model to the return map by its label
-return the map to return

Ultimately I don't want us to hit the schema for field describes twice if we don't have to.